### PR TITLE
`MetaBackend`

### PR DESCRIPTION
**Pull request by BrunoLiegiBastonLiegi**
This implements a `MetaBackend` object that takes care of loading and listing the available `qibotn` backends.

### CODE REVIEW

**Comment by liweintu** [comment]
Thanks @BrunoLiegiBastonLiegi for refactoring the concrete backends into a generic object for loading.

@Vinitha-balachandran , @alecandido Could you help test the refactored object being able to pass the test cases?

**Comment by alecandido** [comment]
> @Vinitha-balachandran , @alecandido Could you help test the refactored object being able to pass the test cases?

I would certainly do it, since the great work of @BrunoLiegiBastonLiegi propagating this through all our backends.
But I'd like to do it on top of existing tests (currently, the CI is completely not running).

**Review by alecandido — APPROVED** [review]

**Review by scarrazza — APPROVED** [review]

File: src/qibotn/backends/cutensornet.py [hunk 1]
```diff
@@ -1,19 +1,9 @@
-import cuquantum  # pylint: disable=import-error
```
Comment (Tankya2) [inline src/qibotn/backends/cutensornet.py:1]: Hi @BrunoLiegiBastonLiegi and @scarrazza , this import is necessary, else line71 will fail. (        self.cuquantum = cuquantum)

Comment (alecandido) [reply src/qibotn/backends/cutensornet.py:1]: I believe you meant L61:
https://github.com/qiboteam/qibotn/blob/217a6338bce0120159fc77648ec38d8d894197c2/src/qibotn/backends/cutensornet.py#L61
but the import could be moved close to
https://github.com/qiboteam/qibotn/blob/217a6338bce0120159fc77648ec38d8d894197c2/src/qibotn/backends/cutensornet.py#L15
for the same reason.

Comment (Tankya2) [reply src/qibotn/backends/cutensornet.py:1]: Yes L61. 
Do you mean to say to move the "import cuquantum" close to line 15?
If so, I am also ok with that suggestion, just that without any import, the code will not run.

**Pull request merged**